File: api/src/main/java/com/example/api/service/MailboxService.java
package com.example.api.service;

import java.util.List;
import com.example.api.model.Mailbox;

public interface MailboxService {

    List<Mailbox> getUserMailboxes(String userId);

    Mailbox addMailbox(Mailbox mailbox, String userId);

    Mailbox updateMailbox(Mailbox mailbox, String userId, Long mailboxId);

    void deleteMailbox(long theId);
}

File: client/src/api/MailboxService.js
import axiosInstance from '../axiosConfig';

export const MailboxService = {
    fetchMailboxes: async (userId) => {
        const response = await axiosInstance.get(`/mailboxes/${userId}`);
        return response.data;
    },

    addMailbox: async (userId, mailboxData) => {
        await axiosInstance.post(`/mailboxes/${userId}`, mailboxData);
    },

    updateMailbox: async (userId, mailboxId, mailboxData) => {
        await axiosInstance.put(`/mailboxes/${userId}/${mailboxId}`, mailboxData);
    },

    deleteMailbox: async (userId, mailboxId) => {
        await axiosInstance.delete(`/mailboxes/${userId}/${mailboxId}`);
    }
};